src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Frame.cs

```diff
@@ -984,6 +984,253 @@ private Task WriteSuffix()
             Output.ProducingComplete(end);
         }
 
+        public bool TakeStartLineSpan(ReadableBuffer buffer, out ReadCursor consumed, out ReadCursor examined)
+        {
+            var start = buffer.Start;
+            var end = buffer.Start;
+
+            examined = buffer.End;
+            consumed = buffer.Start;
+
+            if (_requestProcessingStatus == RequestProcessingStatus.RequestPending)
+            {
+                ConnectionControl.ResetTimeout(_requestHeadersTimeoutMilliseconds, TimeoutAction.SendTimeoutResponse);
+            }
+
+            _requestProcessingStatus = RequestProcessingStatus.RequestStarted;
+
+            var limitedBuffer = buffer;
```

Comment:
Just declare it here but put this assignment in an `else` block, avoiding the potentially wasted assignment.